src/sqlfluff/rules/L005.py

```diff
@@ -43,8 +44,8 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         We need at least one segment behind us for this to work.
 
         """
-        if len(context.raw_stack) >= 1:
-            cm1 = context.raw_stack[-1]
+        if context.raw_segment_pre is not None:
```

Comment:
Switching from raw_stack to `raw_segment_pre` so this rule doesn't require the heavyweight `raw_stack` anymore.

Reply:
Switch to a guard here right ?
```suggestion
        if context.raw_segment_pre is None:
            return
```

Reply:
I would still need a second `return None` because of the nested `if`, so I wonder if this would really be an improvement. I think I'll combine the two `if`s instead, maybe?